iptables-graph

```diff
@@ -45,30 +45,43 @@ def get_node_name(table_name, chain_name):
 def get_port_name(rule_index):
     return "rule_" + str(rule_index)
 
+
+xml_esc = {
+    re.compile('&') : '&amp;' ,
```

Comment:
How about over-long characters in UTF-8 and `<![CDATA[` for XHTML? I think, better to use ready to use library than trying to escape only limited set of characters by handwritten code. Moreover, substitution by multiple regular expressions is very ineffective (as same symbol in each line must be reviewed multiple times).  And such approach is error-prone, as the whole result depend on the orded in which you are applying substitutions. You can construct a single regular expression for this.